website/source/docs/guides/acl-migrate-tokens.html.md

```diff
@@ -0,0 +1,415 @@
+Migrating also enables using the new policy management improvements, stricter
+policy syntax rules and other features of the new system without
+re-issuing all the secrets in use.
+
+The high-level process for migrating a legacy token is as follows:
+
+ 1. Create a new policy or policies that grant the required access
+ 2. Update the existing token to use those policies
+
+### Pre-requisites
+
+This process assumes that the 1.4.0 upgrade is complete including all Legacy
+ACLs having their accessor IDs populated. This might take up to several minutes
+after the servers upgrade in the primary datacenter. You can tell if this is the
+case by using `consul acl token list` and checking that no tokens exist with a
+blank `AccessorID`.
+
+In addition, it is assumed that all clients that might _create_ ACL tokens (e.g.
+Vault's Consul secrets engine) have been updated to use the [new ACL
+APIs](/docs/guides/acl-migrate-tokens.html#api-separation).
+
+Specifically if you are using Vault's Consul secrets engine you need to be
+running Vault 1.0.0 or higher, _and_ you must update all roles defined in Vault
+to specify a list of policy names rather than an inline policy (which causes
+Vault to use the legacy API).
+
+~> **Note:** if you have systems still creating "legacy" tokens with the old
+APIs, the migration steps below will still work, however you'll have to keep
+re-running them until nothing is creating legacy tokens to unsure all tokens are
+migrated.
+
+### Creating Policies
+
+There are several ways to create new policies. The simplest and most automatic
+is to create one new policy for every existing token. This may however result in
+a lot of policies that are logical duplicates and make managing policies harder.
+This approach can be easily accomplished using the [`consul acl policy
+create`](/docs/commands/acl/acl-policy.html#create) command with `-from-token`
+option.
+
+An alternative is to create one policy for each logically distinct token. While
+it's easy enough to do this by hashing the policy contents and only keeping
+distinct hashes, it's hard to extract a meaningful name for the policy that
+expresses it's intent. To make this easier, there is a CLI and API tool that can
+translate a legacy ACL policy into a new ACL policy that is exactly equivalent.
+See [`consul acl translate-rules`](/docs/commands/acl/acl-translate-rules.html).
+
+The final option is to manually inspect all the existing tokens and define named
+policies that describe what the policy is intended to be used for. In this case
+you may also want to modify the policy for example switching to exact-match
+rather than prefix match on resources if that is all that is actually required.
+In this case existing ACL token rules can be inspected using the [`consul acl
+token read -id <accessor_id>`](/docs/commands/acl/acl-token.html#read) command.
+
+### Updating Existing Tokens
+
+Once you have one or more policies that adequately express the rules needed for
```

Comment:
"Once you have created one ...." Can we add created to tie it in more closely with the previous section?